test/setup_environment.sh

```diff
@@ -414,6 +419,35 @@ if [[ -n ${build_jade} ]]; then
     cd ..
 fi
 
+if [[ -n ${build_bitbox02} ]]; then
+    # Clone digital bitbox02 firmware if it doesn't exist, or update it if it does
+    if [ ! -d "bitbox02-firmware" ]; then
+        git clone --recursive https://github.com/BitBoxSwiss/bitbox02-firmware.git
+        cd bitbox02-firmware
+    else
+        cd bitbox02-firmware
+        git fetch
+
+        # Determine if we need to pull. From https://stackoverflow.com/a/3278427
+        UPSTREAM=${1:-'@{u}'}
+        LOCAL=$(git rev-parse @)
+        REMOTE=$(git rev-parse "$UPSTREAM")
+        BASE=$(git merge-base @ "$UPSTREAM")
+
+        if [ $LOCAL = $REMOTE ]; then
+            echo "Up-to-date"
+        elif [ $LOCAL = $BASE ]; then
+            git pull
+        fi
+    fi
+
+    # Build the simulator. This is cached, but it is also fast
+    docker pull shiftcrypto/firmware_v2:latest
+    # The safe.directory config is so that git commands work. even though the repo folder mounted in
+    # Docker is owned by root, which can be different from the owner on the host.
+    docker run -i --rm --volume ./:/bb02 shiftcrypto/firmware_v2 bash -c "git config --global --add safe.directory /bb02 && cd /bb02 && make -j simulator"
```

Comment:
This script needs to work when run inside a docker container as that is how the tests are suggested to be run locally. IIRC Docker within docker doesn't really work, so I don't think this will.